### DISCUSSION TITLE
Add status-info about out-of-memory

### DISCUSSION THREAD

**Discussion by youngjun0627** [lablup/contribution-academy-2021#98] [Show and tell]
- Issue
[#265](https://github.com/lablup/backend.ai/issues/265)

- Solution
docker API를 이용해서 container상태를 확인하고 'OOM'이 발생하면 status-info를 수정하였습니다. 그러나 OOM killer는 프로세스를 종료시키고 container종료를 하지 않습니다. 그러나 OOM이 발생할 때 프로세스 우선순위에 따라 프로세스가 종료되기 때문에 deep-learning에서 매우 치명적인 요소로 작용합니다. 따라서 해당 이슈는 discuss를 통해 해결방안을 모색해야 될 필요가 있습니다. 해당 이슈 관련 [PR](https://github.com/lablup/backend.ai-agent/pull/304)입니다.
(김준기 & 신정규 멘토님의 의견에 따라서 사용자가 recovery를 할 수 있도록 아이디어를 제시할 필요가 있다고 제시하였습니다. 저의 경우 OOM이 발생할 때 해당 container를 종료 시키거나 running상태를 유지하고 status-info를 수정하는 것을 제안했습니다. 다른 의견이 있으시다면 댓글로 달아주세요)

**Comment by lizable** [comment]
Killing container seems reasonable, but we also need to think about in "multi-container" or "multi-node" situation as well. Any ideas for those kinds of situations?

**Reply by youngjun0627** [reply]
I suggested a solution.

- Solution
when OOM event occurs, container is paused(of course, status-info is 'out-of-memory').
The user checks the sesision status-info, and decides to unpause or restart the container after the problem is solved.
So, if we implement this, we also add a command option(unpause) and a container status('PAUSED')
